contrib/http_util/src/test/java/io/opencensus/contrib/http/util/testing/FakeSpan.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http.util.testing;
+
+import io.opencensus.common.Internal;
+import io.opencensus.trace.Annotation;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.EndSpanOptions;
+import io.opencensus.trace.Link;
+import io.opencensus.trace.MessageEvent;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.Status;
+import java.util.Map;
+
+/** A fake {@link Span} which allows user to set {@link SpanContext} upon initialization. */
+@Internal
+public class FakeSpan extends Span {
```

Comment:
Why is this needed?

Reply:
Testing is more simpler with FakeSpan then mocking. In order to test tracecontext injection you would need to mock Span, SpanContext and several methods. I see that we have used TestSpan for log correlation.